examples/hnas/example_mlp.py

```diff
@@ -0,0 +1,272 @@
+        task_df = task_df.assign(task_id=task_id,
+                                 runtime=compute_runtime(task_df, start_timestamp),
+                                 error=compute_error(task_df),
+                                 target_epoch=task_df["epoch"].iloc[-1])
+        task_dfs.append(task_df)
+
+    result = pd.concat(task_dfs, axis="index", ignore_index=True, sort=True)
+
+    # re-order by runtime
+    result = result.sort_values(by="runtime")
+
+    # calculate incumbent best -- the cumulative minimum of the error.
+    result = result.assign(best=result["error"].cummin())
+
+    result.to_csv("history.csv")
+
+
+# CLI
+def parse_args():
+    parser = argparse.ArgumentParser(description='Runs autogluon to optimize '
+                                                 'the hyperparameters of a simple MLP of MNIST ')
+    parser.add_argument('--num_trials', default=10, type=int,
+                        help='number of trial tasks. It is enough to either set num_trials or timout.')
+    parser.add_argument('--timeout', default=1000, type=int,
+                        help='runtime of autogluon in seconds. It is enough to either set num_trials or timout.')
+    parser.add_argument('--num_gpus', type=int, default=0,
+                        help='number of GPUs available to a given trial.')
+    parser.add_argument('--num_cpus', type=int, default=2,
+                        help='number of CPUs available to a given trial.')
+    parser.add_argument('--hostfile', type=argparse.FileType('r'))
+    parser.add_argument('--store_results_period', type=int, default=100,
+                        help='If specified, results are stored in intervals of '
+                             'this many seconds (they are always stored at '
+                             'the end)')
+    parser.add_argument('--scheduler', type=str, default='hyperband_promotion',
+                        choices=['hyperband_stopping', 'hyperband_promotion'],
+                        help='Asynchronous scheduler type. In case of doubt leave it to the default')
+    parser.add_argument('--reduction_factor', type=int, default=3,
+                        help='Reduction factor for successive halving')
+    parser.add_argument('--brackets', type=int, default=1,
+                        help='Number of brackets. Setting the number of brackets to 1 means '
+                             'that we run effectively successive halving')
+    parser.add_argument('--min_resource_level', type=int, default=1,
+                        help='Minimum resource level (i.e epochs) on which a configuration is evaluated on.')
+    parser.add_argument('--searcher', type=str, default='bayesopt',
+                        choices=['random', 'bayesopt'],
+                        help='searcher to sample new configurations')
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == "__main__":
+
+    args = parse_args()
+
+    logging.root.setLevel(logging.INFO)
```

Comment:
This did not produce any log output for me. I had to use the line

logging.basicConfig(level=logging.INFO)

Reply:
ok I changed it